src/models/converters/data_decoded.rs

```diff
@@ -73,3 +75,48 @@ impl DataDecoded {
         }
     }
 }
+
+impl DataDecoded {
+    pub(super) fn build_address_info_index(
+        &self,
+        info_provider: &mut impl InfoProvider,
+    ) -> Option<HashMap<String, AddressInfo>> {
+        let mut index = HashMap::new();
+        if self.method == MULTI_SEND {
+        } else {
+            if let Some(parameters) = &self.parameters {
+                for parameter in parameters {
+                    match &parameter.value {
+                        ParamValue::SingleValue(value) => {
+                            if value.len() == 42 && value.starts_with("0x") {
+                                if let Some(address_info) =
+                                    info_provider.full_address_info_search(&value).ok()
```

Comment:
If the address is already known you should not have to search again ... but probably doesn't matter as the `info_provider` handles that

Reply:
I think it does so only for `token_info` actually. In the case of `contract_info` there isn't even an early return if we search for `0x0`, I will look into this.